src/Renraku/ReUncategorizedClassRule.class.st

```diff
@@ -0,0 +1,50 @@
+"
+Check if the class is uncategorized within its package - if yes it should critisized so it will be tagged
+with either a proper tag or a tag called ""Base"" as then the class belongs to the base part of the package.
+"
+Class {
+	#name : #ReUncategorizedClassRule,
+	#superclass : #ReAbstractRule,
+	#classVars : [
+		'DependencyChecker'
+	],
+	#category : #'Renraku-Rules'
+}
+
+{ #category : #testing }
+ReUncategorizedClassRule class >> checksClass [ 
+
+	^true
+]
+
+{ #category : #accessing }
+ReUncategorizedClassRule class >> uniqueIdentifierName [
+	"This number should be unique and should change only when the rule completely change semantics"
+	
+	^'UncategorizedClassRule'
+]
+
+{ #category : #running }
+ReUncategorizedClassRule >> basicCheck: aClass [
+
+	^(aClass category = aClass package name) 
```

Comment:
I think this is too naive implementation. For small packages, there should be no reason to force use categories. The rule should fail only if the package has already some categories but the particular class is not categorized.

I think it should wait on Pharo 8 since it is not a bugfix